src/upgrade.rs

```diff
@@ -0,0 +1,54 @@
+//! Upgrade Mod
+//!
+//! Makes it easier to upgrade your contract by providing a simple interface for upgrading the code and the state of your contract.
+
+use near_sdk::{env, sys, Gas};
+
+use near_sdk::borsh::{BorshDeserialize, BorshSerialize};
+
+/// Upgrade Trait
+pub trait Upgrade {
+    /// upgrade_all - Upgrades the code and the state of the contract
+    fn upgrade_all();
+}
+
+/// Contracts may implement this trait to inject code into Upgrade functions.
+pub trait UpgradeHook {
+    /// Executed before a upgrade call is conducted
+    fn on_upgrade();
+}
```

Comment:
I don't think this function is ever called.

Reply:
The idea was to keep it flexible